crates/relayer/src/chain/ckb4ibc/monitor.rs

```diff
@@ -96,36 +112,25 @@ impl Ckb4IbcEventMonitor {
                 MonitorCmd::Subscribe(tx) => tx.send(self.event_bus.subscribe()).unwrap(),
             }
         }
-        // block here until the counterparty is revealed
-        if let Ok(client_type) = self.rx_internal.recv() {
-            self.counterparty_client_type = client_type;
-        }
-        let result = async {
-            tokio::select! {
-                batch = self.fetch_channel_events() => batch,
-                batch = self.fetch_connection_events() => batch,
-                batch = self.fetch_packet_events() => batch,
-            }
-        }
-        .await;
-
-        self.process_batch(result);
+        self.process_batch(self.fetch_connection_events().await);
```

Comment:
This changes the concurrency workflow sequencely. The three fetching jobs should start before the `await`, you can use `tokio::join!`.

Example:

``` rust
        let futs = 
            tokio::join! (
                self.fetch_channel_events(),
                self.fetch_connection_events(),
                self.fetch_packet_events(),
            )
```

Reply:
My understanding is that the fetch functions will only try to search once and will return an empty batch if no (new) events are found, so running sequentially is OK. (Join is also OK, select will be a mistake.)

Reply:
they use `.await` but won't block if there is no matches, and I consider to proceed them in a sequence of connection, channel and packet, I decide to invoke them sequentially

Reply:
All the three fetch function called `search_and_extract`, which request to CKB to fetch cells, I see no reason why we not call these function concurrently.